code/modules/mob/living/simple_animal/revenant/revenant_abilities.dm

```diff
@@ -185,22 +117,71 @@
 		spawn(0)
 			if(T.flags & NOJAUNT)
 				T.flags -= NOJAUNT
-			for(var/obj/machinery/bot/bot in T.contents)
-				bot.locked = 0
-				bot.open = 1
-				bot.emagged = 2
-				bot.Emag(null)
+			if(!istype(T, /turf/simulated/wall/cult) && istype(T, /turf/simulated/wall) && prob(30))
+				T.ChangeTurf(/turf/simulated/wall/cult)
+			if(!istype(T, /turf/simulated/floor/engine/cult) && istype(T, /turf/simulated/floor) && prob(30))
+				T.ChangeTurf(/turf/simulated/floor/engine/cult)
 			for(var/mob/living/carbon/human/human in T.contents)
 				human << "<span class='warning'>You suddenly feel tired.</span>"
-				human.adjustStaminaLoss(50)
+				human.adjustStaminaLoss(35)
+			for(var/obj/structure/window/window in T.contents)
+				window.hit(rand(50,125))
+			for(var/obj/machinery/light/light in T.contents)
+				light.flicker() //spooky
+	user.reveal(reveal)
+	user.stun(stun)
+
+
+//Malfunction: Makes bad stuff happen to robots and machines.
+/obj/effect/proc_holder/spell/aoe_turf/revenant_malf
+	name = "Malfunction (50E)"
+	desc = "Corrupts and damages nearby machines and mechanical objects."
+	panel = "Revenant Abilities (Locked)"
+	charge_max = 150
+	clothes_req = 0
+	range = 1
+	var/reveal = 60
+	var/stun = 30
+	var/locked = 1
+
+/obj/effect/proc_holder/spell/aoe_turf/revenant_malf/cast(list/targets, mob/living/simple_animal/revenant/user = usr)
+	if(locked)
+		if(!user.castcheck(-35))
+			charge_counter = charge_max
+			return
+		user << "<span class='info'>You have unlocked Malfunction!</span>"
+		name = "Malfunction (15E)"
+		panel = "Revenant Abilities"
+		locked = 0
+		range = 4
+		charge_counter = charge_max
+		return
+	if(!user.castcheck(-15))
+		charge_counter = charge_max
+		return
+	for(var/turf/T in targets)
+		spawn(0)
+			for(var/obj/machinery/bot/bot in T.contents)
+				if(!bot.emagged)
+					bot.locked = 0
+					bot.open = 1
+					bot.Emag(null)
 			for(var/mob/living/silicon/robot/robot in T.contents)
 				robot.visible_message("<span class='warning'>[robot] lets out an alarm!</span>", \
 									  "<span class='boldannounce'>01001111 01010110 01000101 01010010 01001100 01001111 01000001 01000100</span>") //Translates to "OVERLOAD"
 				robot << 'sound/misc/interference.ogg'
 				robot.emp_act(2)
 				playsound(robot, 'sound/machines/warning-buzzer.ogg', 50, 1)
-			for(var/obj/structure/window/window in T.contents)
-				window.hit(rand(50,200))
-			for(var/obj/machinery/light/light in T.contents)
-				light.flicker() //spooky
-	user.reveal(100)
+			for(var/mob/living/carbon/human/human in T.contents)
```

Comment:
yes but Malfunction is 50E and 15E per use, by that point the revenant has earned the ability to kill the AI, in my eyes, and really, besides being a dick, it has no need to (as the AI can't see the revenant unless people can, and people are the only thing it can drain)

plus this is a very snowflakey almost copy-pasta implementation of EMP effects at this point.